CRM/Civicase/Event/Listener/CaseTypeCategoryIsActiveToggler.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+use CRM_Civicase_Factory_CaseTypeCategoryEventHandler as CaseTypeCategoryEventHandlerFactory;
+use Civi\API\Event\RespondEvent;
+
+/**
+ * Class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler.
+ *
+ * Contains extra processing for case type category when it is
+ * enabled / disabled, e.g. show / hide related menu item.
+ */
+class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler {
+
+  /**
+   * Update case type category related objects on category update.
+   *
+   * @param \Civi\API\Event\RespondEvent $e
+   *   Event data.
+   */
+  public static function onRespond(RespondEvent $e) {
+    $apiRequest = (array) $e->getApiRequest();
```

Comment:
The doc says that the returned value should be an array, so normally we shouldn't need to type cast this. Out of curiosity, what is the returned object class?

Reply:
No idea. Normally it returns an array, so not sure why there was an object on site deployment.

Reply:
That is the more reason why we need to find out why and what Object it returns. Sometimes we need to find out why some things happen so that we can fix, report to core or guard against it next time.

Reply:
How can I debug site deployment? Do you really think we should spent time for this?
Maybe we could just add one more check to the `if` on the next line?
```if (is_array($api_Request) && ...)```

Reply:
Try debug on local first using Xdebug to inspect the object it returns. Shouldn't take long at all.